std/range/package.d

```diff
@@ -4196,6 +4209,12 @@ auto iota(E)(E end)
 // Specialization for floating-point types
 auto iota(B, E, S)(B begin, E end, S step)
 if (isFloatingPoint!(CommonType!(B, E, S)))
+in
+{
+    assert(step != 0, "iota: step must not be 0");
+    assert((end - begin) / step >= 0, "iota: incorrect startup parameters");
```

Comment:
AFAICS, type `S` can easily be a floating-point number, which could make these equality comparisons flaky. How close to 0 is illegal, and is it really OK that different floating-point types for `S` change the margin? Does comparing to 0 only check the two special values +0.0 and -0.0? Or is it just +0.0?